public/app/features/dashboard/components/TransformationsEditor/TransformationFilter.tsx

```diff
@@ -0,0 +1,55 @@
+import { css } from '@emotion/css';
+import React, { useMemo } from 'react';
+
+import {
+  DataFrame,
+  DataTransformerConfig,
+  GrafanaTheme2,
+  StandardEditorContext,
+  StandardEditorsRegistryItem,
+} from '@grafana/data';
+import { useStyles2 } from '@grafana/ui';
+import { FrameSelectionEditor } from 'app/plugins/panel/geomap/editor/FrameSelectionEditor';
+
+interface TransformationFilterProps {
+  index: number;
+  config: DataTransformerConfig;
+  data: DataFrame[];
+  onChange: (index: number, config: DataTransformerConfig) => void;
+}
+
+export const TransformationFilter = ({ index, data, config, onChange }: TransformationFilterProps) => {
+  const styles = useStyles2(getStyles);
+  const context = useMemo(() => {
+    // eslint-disable-next-line
+    return { data } as StandardEditorContext<unknown>;
+  }, [data]);
+
+  return (
+    <div className={styles.wrapper}>
+      <h5>Apply tranformation to</h5>
+      <FrameSelectionEditor
```

Comment:
This component is used in gemap to pick a single frame.  This is sufficient to get started, but note that the config object is any frame matcher, so we will eventually need something (in both places) that is more robust